syntax = "proto3";

import "google/protobuf/any.proto";

import "inviteRefusalReason.proto";

message LobbyCommand {
	google.protobuf.Any command = 1;
}

message CommandCreateGame {
	string description = 1;
}

message CommandJoinGame {
	sint32 game_id = 1;
}

message CommandEnterLobby {}
message CommandEnterQueue {}
message CommandLeaveQueue {}

message CommandInviteToPlay {
    sint32 user_id = 1;
}

message CommandCancelInvite {}
message CommandDeclineInvite {
    sint32 inviter_id = 1;
    InviteRefusalReason reason = 2;
}

message CommandAcceptInvite {
    sint32 inviter_id = 1;
}
